src/ROUTES.ts

```diff
@@ -265,12 +265,6 @@ const ROUTES = {
         route: 'r/:reportID/invite',
         getRoute: (reportID: string) => `r/${reportID}/invite` as const,
     },
-
-    // To see the available iouType, please refer to CONST.IOU.TYPE
-    MONEY_REQUEST: {
-        route: ':iouType/new/:reportID?',
-        getRoute: (iouType: string, reportID = '') => `${iouType}/new/${reportID}` as const,
-    },
```

Comment:
Hey! I've added the mentioned code, and I don't have an instant answer to the problem, but I can explain what is the intention of the code in question. 

We can run NewDot as a part of another native app (so called OldDot). In order to open it there, we can pass a url treated as a deeplink to NewDot.  Nevertheless some urls are generated dynamically, because they add eg. ids to the path, exactly like in `/request/new/scan`. And we need to have NewDot running for that. 

Therefore the flow looks like this:

We open NewDot with a url as an argument. After the app booted, we pass the url to `parseHybridAppUrl` function, and it returns dynamically generated url that we can pass to the `Navigation.navigate()` function as `Route`. 

For some reason TS thinks, that you can pass a `HybridAppRoute` to `Navigation.navigate()`, even though `parseHybridAppUrl` explicitly returns `Route`. I don't know why it happened, but I hope that my explanation will let you understand how the function in question should behave 😄

Reply:
> @staszekscp I think it's because condition `NativeModules.HybridAppModule`, TS is unsure if it's true in all cases so if it's false it's not only `Route`, but also `Route | /request/new/scan`
> 
> Is it right if I assume, if `NativeModules.HybridAppModule` is falsy, the `exitTo` would be a `Route` type?

Yes, the assumption is correct!